### PR TITLE
Feat: donation

### PR DESCRIPTION
**Pull request by nachomazzara**
## Donation contract

A simple contract which mints NFTs when someone sends a donation.

The NFT minted should be a DCL collection.

The minting will be done by defined a specific wearable id.

The contract should be initialized with:
- _**_fundsRecipient**_: The address that should receive the funds.
- _**_erc721Collection**_: NFT collection address.
- _**_minDonation**_: Minimum donation accepted to mint an NFT. Otherwise should call the method to donate in exchange for nothing.
- _**_maxNFTsPerCall**_: Maximum amount of NFTs issued per call.

### CODE REVIEW

File: contracts/Donation.sol [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    function donate() external payable {
+        require(msg.value > 0, "The donation should be higher than 0");
+
+        fundsRecipient.transfer(msg.value);
+
+        emit Donated(msg.sender, msg.value);
+    }
+
+     /**
+     * @dev Donate in exchange of a random NFT.
```
Comment (nachomazzara) [inline contracts/Donation.sol]: ``
```suggestion
     * @dev Donate in exchange for a random NFT.
```

File: contracts/Donation.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     */
+    function donateForNFT() external payable {
+        require(msg.value >= minDonation, "The donation should be higher or equal than the minimum donation ETH");
+        require(issued < maxIssuance, "All wearables have been minted");
+
+        uint256 optionToMint;
+        if (lastOptionIssued == 0 && issued == 0) {
+            optionToMint = 0;
+        } else {
+            optionToMint = (++lastOptionIssued % maxOptions);
```
[the remark this answers is not in the document]
Comment (nachomazzara) [reply contracts/Donation.sol]: Yes! that is a really good observation but, as we are going to use a controlled number of items this time, that makes the contract a little bit expensive and that is why I removed it.

**Comment by agusferreira** [comment]
LGTM

**Pull request merged**